### PR TITLE
CSV Generator

### PR DESCRIPTION
**Pull request by f19ps**
- Added dependencies: `rails`, `activeadmin` and `sidekiq`.
- This PR allows us to include (in the admin views) something like this:
```
csv_report(columns: {
  id: :id,
  title: :title,
  version: :version,
  date: :created_at
}, decorate_model: false)
```

- In this initial version we cannot add a complex value, for example: `file_status:  "file.url ? 'ready' : 'pending'"`. This will be included in the following versions.
- At the moment, the files are stored in the temporal folder of the Rails project.

### CODE REVIEW

File: lib/activeadmin_async_exporter/reports/dsl.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      def csv_report(columns:, decorate_model: false)
+        action_item :download_csv, only: :index do
+          link_to 'Download CSV',
+            { action: :download_csv, params: params },
+            { method: :post, data: { confirm: 'Are you sure you want to generate this report?' } }
+        end
+
+        collection_action :download_csv, method: :post do
+          options = {
+            controller: self.class.to_s,
```
Comment (horacio) [inline lib/activeadmin_async_exporter/reports/dsl.rb]: We could use `self.class.name` here

File: lib/activeadmin_async_exporter/reports/worker.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    include Sidekiq::Worker
+
+    def perform(options = {})
+      controller = options['controller'].classify.constantize.new
+      columns = options['columns']
+
+      headers = columns.keys
+      evaluators = columns.values
+
+      path = "#{Rails.root.to_s}/tmp/#{filename(controller)}"
```
Comment (horacio) [inline lib/activeadmin_async_exporter/reports/worker.rb]: We could also say `Rails.root.join('tmp', filename(controller))`

File: lib/activeadmin_async_exporter/reports/worker.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      
+      CSV.open(path, 'wb', headers: true) do |csv|
+        csv << headers.map { |key| key.to_s }
```
Comment (horacio) [inline lib/activeadmin_async_exporter/reports/worker.rb]: We could use `csv << headers.map(&:to_s)` here

File: lib/activeadmin_async_exporter/reports/worker.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        collection(controller, options).find_in_batches do |group|
+          group.each do |m|
+            m = m.decorate if options['decorate_model']
+            values = []
+            evaluators.each do |ev|
+              values << m.send(ev)
+            end
+            csv << values
```
Comment (horacio) [inline lib/activeadmin_async_exporter/reports/worker.rb]: We could also do `csv << evaluators.collect { |ev| m.send(ev) }`

File: lib/activeadmin_async_exporter/reports/worker.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    private
+
+    def collection(controller, options)
+      controller.scoped_collection.ransack(options['query']).result
+    end
+
+    def filename(controller)
+      [controller.scoped_collection.name.pluralize.try(:downcase), Time.current.try(:to_i)].join('_') + '.csv'
```
Comment (horacio) [inline lib/activeadmin_async_exporter/reports/worker.rb]: I think we can call `downcase` and `to_i` safely here.

**Review by horacio — COMMENTED** [review]
@fpariani Hey! Really cool project. I'd love to help with anything you need.

**Comment by f19ps** [comment]
> @fpariani Hey! Really cool project. I'd love to help with anything you need.

@horacio Really nice comments! Sure, we can work together!

**Review by santiagovidal — APPROVED** [review]

**Review by horacio — APPROVED** [review]

**Pull request merged**